src/flask/routes/oauth2.py

```diff
@@ -0,0 +1,51 @@
+from flask import Blueprint, request, make_response
+
+import jwt
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.exceptions.exceptions import UnAuthorizedEndpoint
+from src.lib.repositories.impl_v2.oauth2_repository_impl import Oauth2RepositoryImpl
+from src.utils.oauth2_util import validate_roles_and_scopes
+
+
+def setup_oauth2_routes(ioc):
+
+    oauth2_blueprint = Blueprint("oauth2_blueprint", __name__)
+    oauth2_repository = ioc.get(Oauth2RepositoryImpl)
+
+    @oauth2_blueprint.before_app_request
+    def before_request_middleware():
+        endpoint_url = request.url_rule
+        endpoint_method = request.method
+        endpoint = str(endpoint_url) + ", " + str(endpoint_method)
+        env_config = get_env_config_instance()
+
+        try:
+            access_token = request.args.get("access_token")
+            oauth2_repository.validate_token(access_token)
+            validate_roles_and_scopes(
```

Comment:
Send the whole `request` object here let `validate_roles_and_scopes` build what it needs from the request

Reply:
done.